diff --git a/projectx-backend/src/main/java/com/tamantaw/projectx/backend/common/exceptionHandlers/MVCExceptionHandler.java b/projectx-backend/src/main/java/com/tamantaw/projectx/backend/common/exceptionHandlers/MVCExceptionHandler.java
index a858f2e0f67547c6c3eda37c60e05f46f980946c..a039d7e8db2aa2cc012592e2b6303f61fb1cef83 100644
--- a/projectx-backend/src/main/java/com/tamantaw/projectx/backend/common/exceptionHandlers/MVCExceptionHandler.java
+++ b/projectx-backend/src/main/java/com/tamantaw/projectx/backend/common/exceptionHandlers/MVCExceptionHandler.java
@@ -218,52 +218,55 @@ public class MVCExceptionHandler {
 		pd.setType(URI.create(typeUri));
 		pd.setProperty("timestamp", Instant.now());
 		pd.setProperty("path", request.getRequestURI());
 		return pd;
 	}

 	private boolean isApiRequest(HttpServletRequest request) {
 		return request.getRequestURI().startsWith("/api/")
 				|| "XMLHttpRequest".equals(request.getHeader("X-Requested-With"));
 	}

 	private void logError(String handler, Exception e) {
 		errorLogger.error(LoggerConstants.LOG_BREAKER_OPEN);
 		errorLogger.error("Handler ==> {}", handler);
 		errorLogger.error(e.getMessage(), e);
 		errorLogger.error(LoggerConstants.LOG_BREAKER_CLOSE);
 	}

 	private ModelAndView errorView(HttpStatus status,
 	                               String view,
 	                               Authentication auth,
 	                               HttpServletRequest request) {

 		ModelAndView mav = new ModelAndView();

+		String normalizedView = normalizeErrorView(view);
+
 		// ✅ THIS IS CRITICAL
-		mav.setViewName(normalizeErrorView(view));
+		mav.setViewName("fragments/layouts/error/template");
+		mav.addObject("view", normalizedView);
 		mav.setStatus(status);

 		mav.addObject("referer", request.getHeader("referer"));

 		mav.addObject("projectVersion", BaseMVCController.getProjectVersion());
 		mav.addObject("buildNumber", BaseMVCController.getBuildNumber());
 		mav.addObject("appShortName", BaseMVCController.getAppShortName());
 		mav.addObject("appFullName", BaseMVCController.getAppFullName());
 		mav.addObject("pageName", "Error!");

 		mav.addObject("isProduction",
 				Arrays.asList(environment.getActiveProfiles()).contains("prd"));

 		return mav;
 	}

 	private String normalizeErrorView(String view) {
 		if (view == null) {
 			return null;
 		}

 		// accept "/error/404" or "error/404"
 		int idx = view.indexOf("/error/");
 		return (idx >= 0) ? view.substring(idx + 1) : view;
 	}
