diff --git a/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/repository/base/AbstractRepositoryImpl.java b/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/repository/base/AbstractRepositoryImpl.java
index 752be83eae9c2db0ae839592d458bfc1843be879..19a22605f0d5ea05b4cb55ac01e40dc1d5523d0d 100644
--- a/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/repository/base/AbstractRepositoryImpl.java
+++ b/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/repository/base/AbstractRepositoryImpl.java
@@ -390,72 +390,76 @@ public abstract class AbstractRepositoryImpl<
 		Assert.notNull(criteria, "Criteria must not be null");

 		@SuppressWarnings("unchecked")
 		EntityPathBase<E> typedPath =
 				(EntityPathBase<E>) path;

 		List<Long> ids = findIds(criteria);
 		if (ids.isEmpty()) {
 			return 0;
 		}

 		long affected = 0;

 		for (List<Long> chunk : chunk(ids, bulkInChunkSize())) {

 			JPAUpdateClause update =
 					queryFactory.update(typedPath);

 			applyAudit(update, updatedBy);
 			spec.apply(update, typedPath);

 			affected +=
 					update.where(audit.id.in(chunk)).execute();
 		}

+		entityManager.clear();
+
 		return affected;
 	}

 	@Override
 	public long deleteByCriteria(CRITERIA criteria) {

 		Assert.notNull(criteria, "Criteria must not be null");

 		List<Long> ids = findIds(criteria);
 		if (ids.isEmpty()) {
 			return 0;
 		}

 		long affected = 0;

 		for (List<Long> chunk : chunk(ids, bulkInChunkSize())) {
 			affected += queryFactory
 					.delete(path)
 					.where(audit.id.in(chunk))
 					.execute();
 		}

+		entityManager.clear();
+
 		return affected;
 	}

 	// ----------------------------------------------------------------------
 	// INTERNAL HELPERS
 	// ----------------------------------------------------------------------

 	protected AbstractJPAQuery<?, ?> createQuery(
 			Predicate predicate,
 			String... hints) {

 		AbstractJPAQuery<?, ?> query =
 				querydsl.createQuery(path);

 		if (predicate != null) {
 			query.where(predicate);
 		}

 		QueryHints qh = getRelatedDataHints(hints);
 		if (qh != null) {
 			qh.forEach(query::setHint);
 		}

 		return query;
 	}
