diff --git a/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/mapper/AdministratorLoginHistoryMapper.java b/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/mapper/AdministratorLoginHistoryMapper.java
index 3ec17d27ea099e7486107b8ba730baf5f831cc69..df45ffb4750bb24f202423b70c8282281e62b99c 100644
--- a/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/mapper/AdministratorLoginHistoryMapper.java
+++ b/projectx-persistence/src/main/java/com/tamantaw/projectx/persistence/mapper/AdministratorLoginHistoryMapper.java
@@ -1,47 +1,57 @@
 package com.tamantaw.projectx.persistence.mapper;

 import com.tamantaw.projectx.persistence.config.MapStructConfig;
+import com.tamantaw.projectx.persistence.dto.AdministratorDTO;
 import com.tamantaw.projectx.persistence.dto.AdministratorLoginHistoryDTO;
 import com.tamantaw.projectx.persistence.entity.AdministratorLoginHistory;
 import com.tamantaw.projectx.persistence.mapper.base.AbstractMapper;
 import com.tamantaw.projectx.persistence.mapper.base.MappingContext;
 import org.hibernate.Hibernate;
+import org.hibernate.proxy.HibernateProxy;
 import org.mapstruct.*;

 @Mapper(
 		config = MapStructConfig.class
 )
 public interface AdministratorLoginHistoryMapper
 		extends AbstractMapper<AdministratorLoginHistoryDTO, AdministratorLoginHistory> {

 	@Override
 	@Mapping(target = "administrator", ignore = true)
 	AdministratorLoginHistory toEntity(AdministratorLoginHistoryDTO dto);

 	@Override
 	@Mapping(target = "administrator", ignore = true)
 	AdministratorLoginHistoryDTO toDto(
 			AdministratorLoginHistory entity,
 			@Context MappingContext ctx
 	);

 	@AfterMapping
 	default void fillRelations(
 			AdministratorLoginHistory entity,
 			@MappingTarget AdministratorLoginHistoryDTO dto,
 			@Context MappingContext ctx
 	) {
 		if (entity == null || ctx == null) {
 			return;
 		}

-		if (entity.getAdministrator() != null
-				&& Hibernate.isInitialized(entity.getAdministrator())) {
+		boolean administratorInitialized =
+				Hibernate.isPropertyInitialized(entity, "administrator")
+						&& Hibernate.isInitialized(entity.getAdministrator());
+
+		if (administratorInitialized && entity.getAdministrator() != null) {

 			dto.setAdministrator(
 					ctx.getAdministratorMapper()
 							.toDto(entity.getAdministrator(), ctx)
 			);
 		}
+		else if (entity != null && entity.getAdministrator() instanceof HibernateProxy hibernateProxy) {
+			AdministratorDTO administratorDTO = new AdministratorDTO();
+			administratorDTO.setId(hibernateProxy.getHibernateLazyInitializer().getIdentifier());
+			dto.setAdministrator(administratorDTO);
+		}
 	}
 }
